dex/src/main/scala/com/wavesplatform/dex/api/http/routes/v0/HistoryRoute.scala

```diff
@@ -0,0 +1,266 @@
+package com.wavesplatform.dex.api.http.routes.v0
+
+import akka.actor.ActorRef
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server._
+import akka.stream.Materializer
+import akka.util.Timeout
+import com.wavesplatform.dex._
+import com.wavesplatform.dex.actors.address.AddressActor
+import com.wavesplatform.dex.actors.address.AddressActor.OrderListType
+import com.wavesplatform.dex.api.http.directives.HttpKamonDirectives._
+import com.wavesplatform.dex.api.http.directives.ProtectDirective
+import com.wavesplatform.dex.api.http.entities._
+import com.wavesplatform.dex.api.http.protocol.HttpCancelOrder
+import com.wavesplatform.dex.api.http.{HasStatusBarrier, _}
+import com.wavesplatform.dex.api.routes.PathMatchers.{AddressPM, AssetPairPM, PublicKeyPM}
+import com.wavesplatform.dex.api.routes.{ApiRoute, AuthRoute}
+import com.wavesplatform.dex.app.MatcherStatus
+import com.wavesplatform.dex.domain.account.Address
+import com.wavesplatform.dex.domain.asset.AssetPair
+import com.wavesplatform.dex.domain.order.Order.Id
+import com.wavesplatform.dex.domain.utils.ScorexLogging
+import com.wavesplatform.dex.model.{AssetPairBuilder, OrderInfo, OrderStatus}
+import io.swagger.annotations._
+
+import javax.ws.rs.Path
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+
+@Path("/matcher")
+@Api()
+class HistoryRoute(
+  responseTimeout: FiniteDuration,
+  assetPairBuilder: AssetPairBuilder,
+  addressActor: ActorRef,
+  override val matcherStatus: () => MatcherStatus,
+  override val apiKeyHash: Option[Array[Byte]]
+)(implicit mat: Materializer)
+    extends ApiRoute
+    with ProtectDirective
+    with HasStatusBarrier
+    with AuthRoute
+    with ScorexLogging {
+
+  implicit private val executionContext: ExecutionContext = mat.executionContext
+  implicit private val timeout: Timeout = responseTimeout
+
+  override lazy val route: Route = pathPrefix("matcher") {
+    pathPrefix("orderbook") {
+      matcherStatusBarrier {
+        getOrderHistoryByAssetPairAndPKWithSig ~ getOrderHistoryByPKWithSig ~ deleteOrderFromHistoryById
+      }
+    } ~ pathPrefix("orders") {
+      matcherStatusBarrier(getOrderHistoryByAddressWithKey)
+    }
+  }
+
+  private val tupledOrderBookHistoryItem: ((Id, OrderInfo[OrderStatus])) => HttpOrderBookHistoryItem =
```

Comment:
L58:76 can be moved after routes. i think it should slightly improve readability.

Reply:
fixed